### PR TITLE
Feature/doc plugin registration

### PR DESCRIPTION
**Pull request by dereklwood**
updating the plugin documentation for plugin registration

### CODE REVIEW

File: docs/docs/source/guide/admin/chef-automator-plugin.rst [hunk 1]
```diff
@@ -200,7 +207,7 @@ all the nodes in the cluster. It then simply utilizes the community "hostsfile"
 each node.
 
 The example loom service definition invoking this cookbook is called "hosts". It simply sets up a "configure" service
-action of type "chef" and script ``recipe[loom_hosts::default]``. Note that the community "hostsfile" cookbook is not
+action of type "chef" and run_list ``recipe[loom_hosts::default]``. Note that the community "hostsfile" cookbook is not
 needed in the runlist since it is declared in loom_hosts's metadata.
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/chef-automator-plugin.rst]: You hyphenated `run-list` in some places, and used `runlist` in others. Pick one, or standardize on `run_list` everywhere.

File: docs/docs/source/guide/admin/plugins.rst [hunk 2]
```diff
@@ -248,11 +274,90 @@ Additionally, each task can return strings representing ``stdout`` and ``stderr`
...
+Defining Your Plugin
+--------------------
+
+Loom plugins are required to provide a JSON file which defines the plugin, including the following:
+  * plugin name
+  * named providertypes and automatortypes
+  * the class name for each providertype and automatortype
+  * any custom fields required by each providertype and automatortype
+
+Custom fields allow a plugin to announce the fields that it requires.  For example a rackspace provider plugin may require a username and password while the joyent provider plugin can require either a password or a key.  Likewise, the ChefAutomator plugin requires a run-list and JSON attributes, while the ShellAutomator plugin requires a command and arguments.
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugins.rst]: s/rackspace/Rackspace/
s/joyent/Joyent/
s/ChefAutomator/Chef Automator/ # or make them all ChefAutomator
s/ShellAutomator/Shell Automator/ # or make them all ShellAutomator

File: docs/docs/source/guide/admin/plugins.rst [hunk 3]
```diff
@@ -248,11 +274,90 @@ Additionally, each task can return strings representing ``stdout`` and ``stderr`
...
+
+For example, consider the JSON definition file for the Rackspace provider plugin:
+::
+
+    {
+        "name": "rackspace",
+        "description": "Rackspace Publici Cloud provider",
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugins.rst]: Typo

File: docs/docs/source/guide/admin/plugins.rst [hunk 4]
```diff
@@ -248,11 +274,90 @@ Additionally, each task can return strings representing ``stdout`` and ``stderr`
...
+                    ]
+                }
+            }
+        }
+    }
+
+This JSON defines a single plugin which contains a single providertype named "rackspace".  The implementing ruby class for this "rackspace" providertype is "RackspaceProvider".  Additionally, it defines three custom fields which can be set by a Loom administrator: "rackspace_username", "rackspace_api_key", and "rackspace_region".  In this example, they are defined within an "admin" block, which indicates these fields can be set only by a Loom administrator.  Alternatively, defining fields within a "user" block indicates a Loom user can set them when creating a cluster.  Each field has additional elements to specify behavior and describe how the field should be presented to the user on the Loom UI:
+  * field name: will be used as the key in the hash sent to the provisioner plugin.
+  * ``label``: the user-friendly label presented in the Loom UI.
+  * ``type``: must be one of
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugins.rst]: Actually, type can be any of the HTTP input field types, as they are used directly by the UI.

File: docs/docs/source/guide/admin/plugins.rst [hunk 5]
```diff
@@ -248,11 +274,90 @@ Additionally, each task can return strings representing ``stdout`` and ``stderr`
...
+
+    * ``text``: an empty text box
+    * ``password``: a text box which masks the value entered
+    * ``select``: a drop-down menu
+
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugins.rst]: As stated above, any HTTP input field type is valid. The Openstack plugin uses a `checkbox` type for `floating_ip`...

http://www.w3schools.com/tags/att_input_type.asp

File: docs/docs/source/guide/admin/plugins.rst [hunk 6]
```diff
@@ -248,11 +274,90 @@ Additionally, each task can return strings representing ``stdout`` and ``stderr`
...
+  * ``tip``: the text hint displayed as the value before the user enters anything
+  * ``options``: (specific to type "select"), the drop-down menu options
+  * ``default``: (specific to type "select"), the default option.
+  * ``override``: specifies whether or not a field defined in an admin block can be overridden by the user during cluster creation.  default: false
+
+Note that a single plugin may contain multiple providertypes and automatortypes.  The top-level ``providertypes`` and ``automatortypes`` arrays should list them all, and then each will have a corresponding JSON element where the classname and custom fields are defined.
```
Comment (wolf31o2) [inline docs/docs/source/guide/admin/plugins.rst]: s/and then/then/

File: docs/docs/source/guide/admin/chef-automator-plugin.rst (hunk 1 above)
Comment (dereklwood) [inline docs/docs/source/guide/admin/chef-automator-plugin.rst]: run-list is how chef docs refer to it.  run_list is our field.  runlist was the mistake, which i will change to run-list

File: docs/docs/source/guide/admin/plugins.rst (hunk 5 above)
Comment (dereklwood) [inline docs/docs/source/guide/admin/plugins.rst]: so i guess its any HTTP input field type, plus select.

**Comment by wolf31o2** [comment]
:+1: LGTM

**Pull request merged**